api/migrations/0003_fake_data.py

```diff
@@ -211,56 +237,75 @@ def _make_group_recipe(self) -> Recipe:
             status=lambda: self.faker.random_element(GroupStatus.values),
             start_date=self.faker.past_date,
             end_date=self.faker.future_date,
-            # FIXME: Sometimes all of the days are generated as None.
-            #  It violates the model's constraint.
-            #  Just restart the migration and pray to the God of Random.
-            monday=self.__get_random_time_or_none,
-            tuesday=self.__get_random_time_or_none,
-            wednesday=self.__get_random_time_or_none,
-            thursday=self.__get_random_time_or_none,
-            friday=self.__get_random_time_or_none,
-            saturday=self.__get_random_time_or_none,
-            sunday=self.__get_random_time_or_none,
+            **self._get_group_days_of_week(),
         )
 
     def _make_speaking_club_recipe(self) -> Recipe:
-        group_common_recipe = self._make_group_common_recipe(
-            APP_NAME + ".SpeakingClub", teachers_min_value=0, teachers_max_value=2
-        )
+        group_common_recipe = self._make_group_common_recipe(APP_NAME + ".SpeakingClub")
         return group_common_recipe.extend(
             is_for_children=self.faker.pybool,
             language=lambda: self.faker.random_element(Language.objects.all()),
             teachers_under_18=lambda: related(
-                *[self.teacher_under_18_recipe]
-                * self.faker.pyint(min_value=0, max_value=2)
+                *[self.teacher_under_18]
+                * self.faker.pyint(
+                    min_value=MIN_AMOUNT_OF_TEACHERS_UNDER_18_IN_SPEAKING_CLUB,
+                    max_value=MAX_AMOUNT_OF_TEACHERS_UNDER_18_IN_SPEAKING_CLUB,
+                )
             ),
         )
 
+
+class FakeDataPopulator(DataPopulator):
+    def __init__(self, apps: StateApps, schema_editor: DatabaseSchemaEditor) -> None:
+        super().__init__(apps, schema_editor)
+        self.recipes = RecipeStorage()
+
+    def _make_amount_of_recipe_and_skip_database_error(
+        self, recipe: Recipe, amount: int
+    ) -> None:
+        """
+        Ensures that given amount of objects is made.
+
+        While generating objects by recipe, it is possible that some of them will violate any db constraint.
+        We need to recreate such objects to skip db constraints and be sure that needed amount is generated.
+        """
+        for _ in range(amount):
+            while True:
+                try:
+                    recipe.make()
+                    break
+                except DatabaseError:
+                    pass
+
```

Comment:
```suggestion
            while True:
                try:
                    recipe.make()
                except DatabaseError:
                    pass
                else:
                    break

```

Reply:
No difference, of course. Obviously, there can be no error when executing `break`. I'm probably being a maximalist here (only one statement in `try`).

Reply:
I think with `else` it will be less readable. Just imho :)